src/components/types.ts

```diff
@@ -0,0 +1,40 @@
+export interface Product {
+  id: string | number;
+  name?: string;
+  price?: {
+    amount: number;
+    currency: string;
+  };
+  unit?: string;
+  textualAmount?: string;
+  badges?: string;
+  image?: string;
+  preferred?: boolean;
+  pricePerUnit?: {
+    amount: number;
+    currency: string;
+  };
+  sales?: string;
+  packageInfo?: string;
+  inStock?: string;
```

Comment:
Na prvniho pohled si rikam, jestli nemuzeme byt trochu vic striktni u nekterych tech "optional" veci. Musi byt vse optional?

Reply:
taky mam v todo, jde o to, ze obcas fetchuju jen cenu, obcas jen produktove informace a idealne bych k tomu chtela pouzivat jeden typ... ael mozna to rozdelim na vic typů, at je to striktni